File: app/src/main/java/com/example/optiweather/network/WeatherApi.java
package com.example.optiweather.network;

import com.example.optiweather.model.WeatherData;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface WeatherApi {
    @GET("v1/forecast")
    Call<WeatherData> getCurrentWeather(
            @Query("latitude") double latitude,
            @Query("longitude") double longitude,
            @Query("current_weather") boolean currentWeather
    );
}

File: app/src/main/java/com/example/optiweather/network/WeatherApi.kt
package com.example.optiweather.network

import com.example.optiweather.model.WeatherData
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface WeatherApi {
    @GET("v1/forecast")
    fun getCurrentWeather(
        @Query("latitude") latitude: Double,
        @Query("longitude") longitude: Double,
        @Query("current_weather") currentWeather: Boolean
    ): Call<WeatherData?>?
}
